### PR TITLE
feat: Build MSI windows installer; Bundle defguard service with the installer

### PR DESCRIPTION
**Pull request by blazej-teonite**
Windows service is not started after installation. Defguard service exe is located in Defguard's installation folder.

### CODE REVIEW

File: src-tauri/resources-windows/service-fragment.wxs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          Start="auto"
+          ErrorControl="normal"
+        >
+
+        </ServiceInstall>
+        <ServiceControl Id="DefGuardService" Name="DefGuardService" Start="install" Stop="both" Remove="uninstall" /> -->
+      </Component>
+    </DirectoryRef>
+  </Fragment>
+</Wix>
```
Comment (moubctez) [inline src-tauri/resources-windows/service-fragment.wxs:80]: EOL

File: src-tauri/src/bin/defguard-service.rs [hunk 2]
```diff
@@ -47,8 +44,127 @@ async fn main() -> anyhow::Result<()> {
...
+        define_windows_service,
+        service::{
+            ServiceControl, ServiceControlAccept, ServiceExitCode, ServiceState, ServiceStatus,
+            ServiceType,
+        },
+        service_control_handler::{self, ServiceControlHandlerResult},
+        service_dispatcher, Result,
+    };
+
+    const SERVICE_NAME: &str = "DefguardService";
```
Comment (moubctez) [inline src-tauri/src/bin/defguard-service.rs:86]:
```suggestion
    static SERVICE_NAME: &str = "DefguardService";
```

File: src-tauri/src/bin/defguard-service.rs [hunk 3]
```diff
@@ -47,8 +44,127 @@ async fn main() -> anyhow::Result<()> {
...
+    pub fn run() -> Result<()> {
+        // Register generated `ffi_service_main` with the system and start the service, blocking
+        // this thread until the service is stopped.
+        service_dispatcher::start(SERVICE_NAME, ffi_service_main)
+    }
+
+    define_windows_service!(ffi_service_main, service_main);
+
+    pub fn service_main(_arguments: Vec<OsString>) {
+        if let Err(_e) = run_service() {
```
Comment (moubctez) [inline src-tauri/src/bin/defguard-service.rs:98]:
```suggestion
        if let Err(err) = run_service() {
```

File: src-tauri/resources-windows/main.wxs [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+        <Property Id="ARPPRODUCTICON" Value="ProductIcon" />
+        <Property Id="ARPNOREPAIR" Value="yes" Secure="yes" />      <!-- Remove repair -->
+        <SetProperty Id="ARPNOMODIFY" Value="1" After="InstallValidate" Sequence="execute"/>
+
+        <!-- initialize with previous InstallDir -->
+        <Property Id="INSTALLDIR">
+            <RegistrySearch Id="PrevInstallDirReg" Root="HKCU" Key="Software\\{{manufacturer}}\\{{product_name}}" Name="InstallDir" Type="raw"/>
+        </Property>
+
+        <!-- <Property Id="WixSilentExecCmdLine" Value="[INSTALLDIR]resources-windows\binaries\wireguard-installer.exe" Hidden="yes"/>
```
Comment (wojcik91) [inline src-tauri/resources-windows/main.wxs]: remove if not necessary

File: src-tauri/resources-windows/main.wxs [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+        <InstallExecuteSequence>
+            <Custom Action="DeleteUpdateTask" Before='InstallFinalize'>
+                (REMOVE = "ALL") AND NOT UPGRADINGPRODUCTCODE
+            </Custom>
+        </InstallExecuteSequence>
+        {{/if}}
+
+        <SetProperty Id="ARPINSTALLLOCATION" Value="[INSTALLDIR]" After="CostFinalize"/>
+    </Product>
+</Wix>
```
Comment (wojcik91) [inline src-tauri/resources-windows/main.wxs:326]: add newline

File: src-tauri/src/bin/defguard-service.rs [hunk 6]
```diff
@@ -47,8 +44,127 @@ async fn main() -> anyhow::Result<()> {
...
     Ok(())
 }
+
+#[cfg(windows)]
+fn main() -> windows_service::Result<()> {
+    defguard_windows_service::run()
+}
+
+#[cfg(windows)]
+mod defguard_windows_service {
```
Comment (wojcik91) [inline src-tauri/src/bin/defguard-service.rs:69]: move this to a separate file

File: src-tauri/src/service/mod.rs [hunk 7]
```diff
@@ -105,9 +105,23 @@ impl DesktopDaemonService for DaemonService {
...
+            .collect();
+
         // configure interface
         debug!("Configuring new interface {ifname} with configuration: {config:?}");
-        wgapi.configure_interface(&config).map_err(|err| {
+
+        #[cfg(not(windows))]
+        let configure_interface_result = wgapi.configure_interface(&config);
+        #[cfg(windows)]
+        let configure_interface_result = wgapi.configure_interface(&config, &dns);
```
Comment (wojcik91) [inline src-tauri/src/service/mod.rs:122]: Why this way if there's already `wgapi.configure_dns()` below?

File: src/pages/client/clientAPI/clientApi.ts [hunk 8]
```diff
@@ -29,7 +29,7 @@ import {
 async function invokeWrapper<T>(
   command: TauriCommandKey,
   args?: InvokeArgs,
-  timeout: number = 5000,
+  timeout: number = 15000,
```
Comment (wojcik91) [inline src/pages/client/clientAPI/clientApi.ts:32]: 15 seconds seem a bit much. Is that necessary?

Comment (blazej-teonite) [reply src/pages/client/clientAPI/clientApi.ts:32]: Initial implementation of connection on Windows was slow, I will change it to 10 seconds. Would you like it to be shorter than that?

**Pull request merged**